### PR TITLE
Tezos: support tz3 addresses (P256 elliptic curve)

### PR DESCRIPTION
**Pull request by KhalilBellakrid**
This relates to : https://ledgerhq.atlassian.net/browse/LLC-423
Note: Tested with Live and it's working ! 👍

### CODE REVIEW

File: core/src/tezos/TezosLikeAddress.cpp [hunk 1]
```diff
@@ -119,5 +119,43 @@ namespace ledger {
...
+                    std::vector<uint8_t>{static_cast<uint8_t>(0x02 + (pubKey[64] & 0x01))},
+                    std::vector<uint8_t>{pubKey.begin() + 1, pubKey.begin() + 33}
+            );
+            auto sRes = hex::toString(result);
+            switch (curve) {
+                case api::TezosCurve::ED25519 :
+                    if (pubKey.size() != 33) {
+                        throw Exception(api::ErrorCode::INVALID_ARGUMENT, "Invalid ED25519 public key: should be 33 bytes.");
+                    }
+                    return BLAKE::blake2b(std::vector<uint8_t>{pubKey.begin() + 1, pubKey.end()}, 20);
```
Comment (alekece) [inline core/src/tezos/TezosLikeAddress.cpp]: Just to be sure, why are you skipping the first byte ? Can you add a comment to explain this technical choice ?

File: core/src/tezos/TezosLikeAddress.cpp [hunk 2]
```diff
@@ -119,5 +119,43 @@ namespace ledger {
...
+                default :
+                    if (pubKey.size() != 65) {
+                        throw Exception(api::ErrorCode::INVALID_ARGUMENT, "Invalid SECP256k1 or P256 public key: should be 65 bytes.");
+                    }
+                    return BLAKE::blake2b(
+                            vector::concat(
+                                    std::vector<uint8_t>{static_cast<uint8_t>(0x02 + (pubKey[64] & 0x01))},
+                                    std::vector<uint8_t>{pubKey.begin() + 1, pubKey.begin() + 33}
```
Comment (alekece) [inline core/src/tezos/TezosLikeAddress.cpp]: Ditto

File: core/src/tezos/TezosLikeAddress.cpp [hunk 3]
```diff
@@ -119,5 +119,43 @@ namespace ledger {
...
+                            ),
+                            20);
+            }
+        }
+
+        std::vector<uint8_t> TezosLikeAddress::getPrefixFromImplicitVersion(const std::vector<uint8_t> &implicitVersion,
+                                                                            api::TezosCurve curve) {
+            return vector::concat(
+                    std::vector<uint8_t>{implicitVersion.begin(), implicitVersion.end() - 1},
```
Comment (alekece) [inline core/src/tezos/TezosLikeAddress.cpp]: Same question here but about the skip of the last byte

File: core/src/wallet/tezos/TezosLikeAccount2.cpp [hunk 4]
```diff
@@ -307,8 +307,33 @@ namespace ledger {
...
+                                                } else if (xpubConfig == api::TezosConfigurationDefaults::TEZOS_XPUB_CURVE_SECP256K1) {
+                                                    return api::TezosCurve::SECP256K1;
+                                                }
+                                                return api::TezosCurve::P256;
+                                            };
+                                            // Get sender's curve first
+                                            // For KT accounts, it is always ED25519
+                                            auto senderCurve = api::TezosConfigurationDefaults::TEZOS_XPUB_CURVE_ED25519;
+                                            if (senderAddress.find("KT1") != 0) {
+                                                senderCurve = self->getKeychain()->getConfiguration()
```
Comment (alekece) [inline core/src/wallet/tezos/TezosLikeAccount2.cpp]: Is it valid if `std::find` returns `std::string::npos` ?

Comment (KhalilBellakrid) [reply core/src/wallet/tezos/TezosLikeAccount2.cpp]: Here we test that the address prefix is `KT1` to test if it's an originated account

**Pull request merged**